doc/Learn/Tutorials/Markup/HowTo-CustomMarkupProject-Toolkit.md

```diff
@@ -0,0 +1,540 @@
+
+Now we will check [Uno Toolkit](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/getting-started.html) which is a library that can be added to any new or existing Uno solution.
+
+For this sample you can use the how to [Create your own C# Markup with Toolkit](xref:Reference.Markup.HowTo-MarkupProjectToolkit)
+
+For this sample we will cover this controls:
+
+- [NavigationBar](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/NavigationBar.html)  - Represents a specialized app bar that provides layout for AppBarButton and navigation logic.
+
+- [Chip](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/ChipAndChipGroup.html) - Chips are compact elements that represent an input, attribute, or action.
+
+## NavigationBar
+
+The NavigationBar is a user interface component used to provide navigation between different pages or sections of an application.
+
+The navigation bar can include items such as a back button, a page title, and other navigation elements, depending on your application's structure and requirements.
+
+### Changing UI to have the [NavigationBar](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/NavigationBar.html)
+
+- In the Shared Project open the file *MainPage.cs* and change the content to have the NavigationBar.
+
+    # [**C# Markup**](#tab/cs)
+
+    #### C# Markup
+
+    To do so, create a new class file named SecondPage.
+    Then open the class and replace the content for:
+
+    ```csharp
+    namespace MySampleToolkitProject;
+
+    public sealed partial class SecondPage : Page
+    {
+	    public SecondPage()
+	    {
+		    this
+			    .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))
+			    .Content(new StackPanel()
+			    .VerticalAlignment(VerticalAlignment.Center)
+			    .HorizontalAlignment(HorizontalAlignment.Center)
+			    .Children(
+				    new NavigationBar().Content("Title Second Page"),
+				    new Button()
+					    .Content("Go to Main Page")
+					    .Assign(out var navigationButton)
+			    ));
+
+		    navigationButton.Click += (s, e) =>
+		    {
+			    Frame.Navigate(typeof(MainPage));
+		    };
+	    }
+    }
+
+    ```
+
+    Now we can change the content of the MainPage.cs to have the NavigationBar.
+
+    ```csharp
+    new NavigationBar().Content("Title Main Page")
+    ```
+
+    And For have the controls to navigate for the SecondPage we can add a Button for do it.
+
+    ```csharp
+    new Button()
+		.Content("Go to Second Page")
+		.Assign(out var navigationButton),
+    ```
+
+    The main ideia here is to create a button that have Assign it self the output variable named in this case as navigationButton.
+    After that we need to add a Click EventHandler to add the Navigation using the Frame.Navigate. 
+
+    > Notice how simple it is to create an action for the Button's Click event.
```

Comment:
```suggestion
    Now we can change the content of the MainPage.cs to have the NavigationBar.

    ```csharp
    new NavigationBar().Content("Title Main Page")
    ```

    And to have the controls to navigate for the SecondPage we can add a Button to do it.

    ```csharp
    new Button()
		.Content("Go to Second Page")
		.Assign(out var navigationButton),
    ```

    The main idea here is to create a button that assigns itself the output variable named in this case navigationButton.
    After that, we need to add a Click EventHandler to add the Navigation using the Frame.Navigate. 

    > Notice how simple it is to create an action for the Button's Click event.
```